### ISSUE TITLE
forloop.last incorrect if loop has a limit

### ISSUE THREAD

**Issue by paulrobertlloyd** [harttle/liquidjs#67]
I’ve discovered an issue with the `forloop` object when requesting the last value in an array. This works as expected should the loop have no limit, e.g.:

```njk
<!-- if array = [1,2,3,4,5,6] -->
{% for item in array -%}
  {{ item }}{% unless forloop.last %},{% endunless %}
{%- endfor %}
```

Output: `1,2,3,4,5,6` (note, no trailing comma).

However, if I add a limit, `forloop.last` is not applied:

```njk
<!-- if array = [1,2,3,4,5,6] -->
{% for item in array limit: 3 -%}
  {{ item }}{% unless forloop.last %},{% endunless %}
{%- endfor %}
```

Output: `1,2,3,` (note the trailing comma). Expected output: `1,2,3`

I’ve been using the Ruby implementation of Liquid via Jekyll, and this worked as expected. Building a site using [11ty](https://github.com/11ty/eleventy), for which this project is a dependant, I get the result described above.

**Comment by harttle** [comment]
fixes published @3.1.3

**Comment by paulrobertlloyd** [comment]
Fantastic, thank-you @harttle!
